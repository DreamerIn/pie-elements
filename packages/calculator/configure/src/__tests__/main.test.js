import React from 'react';
import Main from '../main';
import { truncateSync } from 'fs';

describe('Render a calculator element', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Main model={{mode: 'basic'}} onModelChanged={() => true}/>)
  })

  it('Creates snapshot using enzyme', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Test for onChangeHandler', () => {
    wrapper.instance().onChangeHandler('scientific');
    const twoChoice = wrapper.state('twoChoice');
    expect(twoChoice).toEqual('scientific');
  })

});