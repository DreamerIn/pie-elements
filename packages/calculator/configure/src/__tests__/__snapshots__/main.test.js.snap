// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a calculator element Creates snapshot using enzyme 1`] = `
<div>
  <TwoChoice
    header="Choose Calculator Type"
    onChange={[Function]}
    one={
      Object {
        "label": "Basic",
        "value": "basic",
      }
    }
    two={
      Object {
        "label": "Scientific",
        "value": "scientific",
      }
    }
    value=""
  />
</div>
`;
